test/smoke/dashCore.js

```diff
@@ -1,18 +1,79 @@
-describe('DashCore', () => {
-  describe('All nodes', () => {
-    // `subversion` from `getNetworkInfo` response.
-    it('should have correct network type');
+const RpcClient = require('@dashevo/dashd-rpc/promise');
+const SSH = require('simple-ssh');
+const fs = require('fs');
+const utils = require('../testUtils');
 
-    // The number of blocks should be almost the same (-3/+3) and block hash of particular
-    // block height should be the same. Using `blocks` and `bestblockhash` from `GetBlockChainInfo`.
-    it('should propagate blocks');
-  });
-  describe('Masternodes', () => {
-    // masternode status
-    it('should be masternodes');
-  });
-  describe('Miners', () => {
-    // waitfornewblock
-    it('should mine blocks');
+const getNetworkConfig = require('../../lib/test/getNetworkConfig');
+
+const networkConfig = getNetworkConfig();
+
+
+networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+  describe(`DashCore ${nodeName}`, () => {
+    const config = {
+      protocol: 'http',
+      user: 'dashrpc',
+      pass: 'password',
+      host: networkConfig.inventory._meta.hostvars[nodeName].public_ip,
+      port: 20002,
+    };
+    const rpc = new RpcClient(config);
+
+    describe('All nodes', () => {
+      it('should have correct network type/subversion from `getNetworkInfo`', async () => {
```

Comment:
I think we should keep specs less technical. "should have correct network type" will be enought.

Reply:
fixed